### PR TITLE
Support promises

### PR DESCRIPTION
**Pull request by buildmaster**
node-cache-manager expects the library to be able to support promises, as such this makes sure memory manager doesn't break if the code using it expects promise support.

Take if you want to :)

### CODE REVIEW

File: lib/MemoryClient.js [hunk 1]
```diff
@@ -1,51 +1,58 @@
+let _Promise
 export default class MemoryClient {
   constructor (options) {
     this.options = options
     this.data = {}
+    _Promise = options.promiseDependency || global.Promise
```
Comment (theogravity) [inline lib/MemoryClient.js]: why not `this._Promise` or at least wrap it around `if(_Promise)`

Comment (buildmaster) [reply lib/MemoryClient.js]: Will do, any preference?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (buildmaster) [inline .prettierrc]: i'll remove this file too

File: lib/MemoryClient.js [hunk 3]
```diff
@@ -1,51 +1,58 @@
...
 
   get (key, options, cb) {
     if (typeof options === 'function') {
       cb = options
     }
-
-    cb(null, this.data[key] || null)
+    return new _Promise((resolve, reject) => {
+      cb = cb || ((err, result) => (err ? reject(err) : resolve(result)))
+      cb(null, this.data[key] || null)
```
Comment (theogravity) [inline lib/MemoryClient.js]: You need to call `resolve()` in the `cb` situation (like call it after `cb()` or the promise will get 'stuck' and consume memory.

File: lib/MemoryClient.js (hunk 1 above)
Comment (theogravity) [reply lib/MemoryClient.js]: Let's just wrap it.

**Comment by theogravity** [comment]
Thanks for the contrib - see comments

**Comment by theogravity** [comment]
Also please make sure the tests pass. I'm ok if you don't want to add tests for the promise, but make sure the existing ones pass.

**Comment by theogravity** [comment]
Since I have not touched this lib in years, I've added you as a maintainer, and I'll be happy to give you npm publish access (let me know your npm username). Feel free to modernize the lib as you wish!

Please publish as a new major version once the changes are in

File: lib/MemoryClient.js (hunk 1 above)
Comment (buildmaster) [reply lib/MemoryClient.js]: I used `this.Promise` as it seemed to make more sense

File: lib/MemoryClient.js (hunk 3 above)
Comment (buildmaster) [reply lib/MemoryClient.js]: I think this should now be resolved, it'll call resolve on the promise and call the callback.

**Comment by buildmaster** [comment]
actually I think this should be closed in favor of using the built in memory story (that I didn't actually know existed 🤦 ) https://github.com/BryanDonovan/node-cache-manager/blob/master/lib/stores/memory.js

**Comment by theogravity** [comment]
No problem. I'll leave the PR open if you think the native one does not fit your use-case. I'm not sure why I ended up rolling my own at the time myself.

**Pull request closed without merging**